index.rst

```diff
@@ -1,61 +1,311 @@
+Here we explore updates to the `MTAOS (xml) interface`_ and overall software architecture to align it with the evolution of the system design.
+The most significant advancement that affect how the MTAOS operates are;
 
-  .. figure:: /_static/filename.ext
-     :name: fig-label
+  - Better understanding of the observatory operation procedures.
+  - Consolidation of the data access interface, through butler gen3, for local data processing of corner wavefront sensors and remote access for full array mode.
+  - Consolidation of the OCS Controlled Pipeline System (`OCPS`_) for remote data processing.
+  - Conversion of the wavefront estimation pipeline (`WEP`_) into a butler gen3 pipeline task.
 
-     Caption text.
+.. _MTAOS: https://ts-mtaos.lsst.io/index.html
+.. _AOS Workshop: https://confluence.lsstcorp.org/x/zzTbAw
+.. _MTAOS (xml) interface: https://ts-xml.lsst.io/sal_interfaces/MTAOS.html
+.. _OCPS: https://dmtn-133.lsst.io
+.. _WEP: https://github.com/lsst-ts/ts_wep
 
-   Run: ``make html`` and ``open _build/html/index.html`` to preview your work.
-   See the README at https://github.com/lsst-sqre/lsst-technote-bootstrap or
-   this repo's README for more info.
+.. _Data-access-changes:
 
-   Feel free to delete this instructional comment.
+Data access changes
+===================
 
-:tocdepth: 1
+The consolidation of data access through butler gen3 represents an important change in the interface and operation of the MTAOS.
+The following is a series of improvements that can be implemented to improve the MTAOS usability.
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+Processing calibration products
+-------------------------------
 
-.. sectnum::
+This is currently done via the ``processCalibrationProducts`` command, which expected a directory containing the raw calibration data.
+Instead, the butler instances hosting the data will also contain processed calibration products suitable for all exposures.
+This will have to be carefully managed by observatory personnel over time, but is a much more suitable approach than leaving the MTAOS in charge of the process.
 
-.. TODO: Delete the note below before merging new content to the master branch.
+Image ingestion
+---------------
 
-.. note::
+As part of the wavefront processing commands, the current version of the MTAOS receives files names and directory location.
+This data is initially ingested to a butler instance to allow them to be ISR corrected and later WEP processed.
+
+Since the data will now be ingested by a different component and will be made available to the MTAOS through butler, there is no more need for a local ingestion.
+
+Simplification of wfs process commands
+--------------------------------------
+
+As mention above, the wfs process commands receives file names and path.
+With the data already ingested in the butler the images can be uniquely identified by the ``visitId`` entry in the database.
```

Comment:
I believe the `visitId` is not enough. You may also need the `raft` and `sensor` because we may not use all the CCDs. We only pick the CCD needed. In addition, you may also need to have the `snap` or `sequence` to assign which exposure you want to use.

Reply:
Another possibility is to let the `MTAOS` to decide what should be the `raft` and `sensor` and other fields internally.

Reply:
I think `raft` and `sensor` can be derived from the sources selected for wavefront sensing. `visitId` maps to each individual exposure so `snap` and `sequence` would be redundant, I guess. `filter` is also derived from `visitId`.

But this is a good point. I will mention in the text.

Reply:
I just realized it should be the `dataId` not `visitId`.. hehehe.. I will updated the text everywhere..

Reply:
The filter is needed to select the source (bright stars). Therefore, we will not use the butler to get this information. The source selection should happen before the butler is ready for the new images.

Reply:
TeWei has a good point here. I think we should allow
await mtaos.cmd_runWEP.set_start(dataId = [xxx, yyy], sensor = 'R22_S11')
If the sensor argument is not given, then it is assumed to be all sensors that have valid wavefront solutions.

also, "mention above" --> "mentioned above"

Reply:
should we instead allow users to specify a configuration in yaml format? this means we can support any kind of future parameters and would allow us to implement new features on the fly, without the need to redo the MTAOS xml.

Reply:
What would the configuration choices be?